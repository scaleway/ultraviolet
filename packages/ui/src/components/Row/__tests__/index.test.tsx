import { shouldMatchEmotionSnapshot } from '@utils/test'
import { describe, test } from 'vitest'
import { Row } from '..'

describe('row', () => {
  test('renders correctly with default props', () =>
    shouldMatchEmotionSnapshot(
      <Row templateColumns="repeat(2, 1fr)">
        <div>First col</div>
        <div>Second col</div>
      </Row>,
    ))

  test('renders correctly with specific gap', () =>
    shouldMatchEmotionSnapshot(
      <Row gap={1} templateColumns="repeat(2, 1fr)">
        <div>First col</div>
        <div>Second col</div>
      </Row>,
    ))

  test('renders correctly with specific align', () =>
    shouldMatchEmotionSnapshot(
      <Row alignItems="center" gap={1} templateColumns="repeat(2, 1fr)">
        <div>First col</div>
        <div>Second col</div>
      </Row>,
    ))

  test('renders correctly with specific padding', () =>
    shouldMatchEmotionSnapshot(
      <Row gap={1} padding="10px" templateColumns="repeat(2, 1fr)">
        <div>First col</div>
        <div>Second col</div>
      </Row>,
    ))

  test('renders correctly with responsive values', () =>
    shouldMatchEmotionSnapshot(
      <Row
        alignItems={{ large: 'end', medium: 'center', small: 'start' }}
        gap={{ large: 3, medium: 2, small: 1 }}
        justifyContent={{ large: 'end', medium: 'center', small: 'start' }}
        padding={{ large: 2, medium: 1, small: 0 }}
        templateColumns={{
          large: '2fr 1fr',
          medium: '1fr 1fr',
          small: '1fr 2fr',
        }}
      >
        <div>First col</div>
        <div>Second col</div>
      </Row>,
    ))
})
