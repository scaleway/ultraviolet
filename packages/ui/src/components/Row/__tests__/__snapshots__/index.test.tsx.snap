// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Row > renders correctly with default props 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__x6hyh50 "
      style="--_3mg8xe0: repeat(2, 1fr); --_3mg8xe1: repeat(2, 1fr); --_3mg8xe2: repeat(2, 1fr); --_3mg8xe3: repeat(2, 1fr); --_3mg8xe4: repeat(2, 1fr); --_3mg8xe5: repeat(2, 1fr);"
    >
      <div>
        First col
      </div>
      <div>
        Second col
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Row > renders correctly with responsive values 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__x6hyh50 styles_alignItems_end_large__x6hyh535 styles_alignItems_center_medium__x6hyh52s styles_alignItems_start_small__x6hyh52x styles_gap_1.5rem_large__x6hyh5n styles_gap_1rem_medium__x6hyh5g styles_gap_0.5rem_small__x6hyh59 styles_justifyContent_end_large__x6hyh55b styles_justifyContent_center_medium__x6hyh54y styles_justifyContent_start_small__x6hyh553"
      style="--_3mg8xe0: 2fr 1fr; --_3mg8xe2: 1fr 1fr; --_3mg8xe3: 1fr 2fr; --_3mg8xe6: 2; --_3mg8xe8: 1;"
    >
      <div>
        First col
      </div>
      <div>
        Second col
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Row > renders correctly with specific align 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__x6hyh50 styles_alignItems_center_xxsmall__x6hyh52p styles_gap_0.5rem_xxsmall__x6hyh57"
      style="--_3mg8xe0: repeat(2, 1fr); --_3mg8xe1: repeat(2, 1fr); --_3mg8xe2: repeat(2, 1fr); --_3mg8xe3: repeat(2, 1fr); --_3mg8xe4: repeat(2, 1fr); --_3mg8xe5: repeat(2, 1fr);"
    >
      <div>
        First col
      </div>
      <div>
        Second col
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Row > renders correctly with specific gap 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__x6hyh50 styles_gap_0.5rem_xxsmall__x6hyh57"
      style="--_3mg8xe0: repeat(2, 1fr); --_3mg8xe1: repeat(2, 1fr); --_3mg8xe2: repeat(2, 1fr); --_3mg8xe3: repeat(2, 1fr); --_3mg8xe4: repeat(2, 1fr); --_3mg8xe5: repeat(2, 1fr);"
    >
      <div>
        First col
      </div>
      <div>
        Second col
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Row > renders correctly with specific padding 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      class="styles__x6hyh50 styles_gap_0.5rem_xxsmall__x6hyh57"
      style="--_3mg8xe0: repeat(2, 1fr); --_3mg8xe1: repeat(2, 1fr); --_3mg8xe2: repeat(2, 1fr); --_3mg8xe3: repeat(2, 1fr); --_3mg8xe4: repeat(2, 1fr); --_3mg8xe5: repeat(2, 1fr); --_3mg8xe6: 10px; --_3mg8xe7: 10px; --_3mg8xe8: 10px; --_3mg8xe9: 10px; --_3mg8xea: 10px; --_3mg8xeb: 10px;"
    >
      <div>
        First col
      </div>
      <div>
        Second col
      </div>
    </div>
  </div>
</DocumentFragment>
`;
